map_gui/src/tools/labels.rs

```diff
@@ -45,29 +72,27 @@ impl DrawRoadLabels {
     }
 
     pub fn draw(&self, g: &mut GfxCtx, app: &dyn AppLike) {
-        let (zoom, idx) = Self::discretize_zoom(g.canvas.cam_zoom);
-        let value = &mut self.per_zoom.borrow_mut()[idx];
-        if value.is_none() {
-            *value = Some(self.render(g, app, zoom));
+        let mut per_zoom = self.per_zoom.borrow_mut();
+        if per_zoom.is_none() {
+            *per_zoom = Some(PerZoom::new(g.canvas.settings.min_zoom_for_detail));
         }
-        g.redraw(value.as_ref().unwrap());
-    }
+        let per_zoom = per_zoom.as_mut().unwrap();
 
-    fn discretize_zoom(zoom: f64) -> (f64, usize) {
-        // TODO Maybe more values between 1.0 and min_zoom_for_detail?
-        if zoom >= 1.0 {
-            return (1.0, 10);
+        let (zoom, idx) = per_zoom.discretize_zoom(g.canvas.cam_zoom);
+        let draw = &mut per_zoom.draw_per_zoom[idx];
+        if draw.is_none() {
+            *draw = Some(self.render(g, app, zoom));
         }
-        let rounded = (zoom * 10.0).round();
-        let idx = rounded as usize;
-        (rounded / 10.0, idx)
+        g.redraw(draw.as_ref().unwrap());
     }
 
     fn render(&self, g: &mut GfxCtx, app: &dyn AppLike, zoom: f64) -> Drawable {
         let mut batch = GeomBatch::new();
         let map = app.map();
 
-        let text_scale = 1.0 + 2.0 * (1.0 - zoom);
```

Comment:
It took an embarrassing amount of tries to figure out what to do here in between canvas zooms of 1 and 4 (the `min_zoom_for_detail`, after which we switch to the more detailed rendering). In the end I realized the text should always be the same size on the screen